core/core-impl/src/main/java/io/fabric8/launcher/core/impl/events/ArtemisStatusMessageEventBroker.java

```diff
@@ -0,0 +1,98 @@
+package io.fabric8.launcher.core.impl.events;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.jms.Destination;
+import javax.jms.JMSConsumer;
+import javax.jms.JMSContext;
+import javax.jms.JMSException;
+import javax.jms.JMSProducer;
+
+import io.fabric8.launcher.base.EnvironmentSupport;
+import io.fabric8.launcher.base.JsonUtils;
+import io.fabric8.launcher.core.api.events.StatusMessageEvent;
+import io.fabric8.launcher.core.api.events.StatusMessageEventBroker;
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory;
+
+/**
+ * A {@link StatusMessageEventBroker} implementation that uses ActiveMQ Artemis
+ *
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public class ArtemisStatusMessageEventBroker implements StatusMessageEventBroker {
+
+    private static final Logger logger = Logger.getLogger(ArtemisStatusMessageEventBroker.class.getName());
+
+    public static final String DESTINATION_NAME = "launcher-status-messages";
+
+    private final Destination destination;
+
+    private final ActiveMQConnectionFactory activeMQConnectionFactory;
+
+    private final JMSContext jmsContext;
+
+    private final Map<UUID, JMSConsumer> consumers = new ConcurrentHashMap<>();
+
+    public ArtemisStatusMessageEventBroker(String url, String user, String password) {
+        activeMQConnectionFactory = new ActiveMQConnectionFactory(url, user, password);
+        jmsContext = activeMQConnectionFactory.createContext();
+        jmsContext.setClientID(EnvironmentSupport.INSTANCE.getEnvVarOrSysProp("HOST_NAME", "localhost"));
```

Comment:
`HOST_NAME`? Or `HOSTNAME`?

Reply:
Thanks, I was unsure of the environment var name used in K8s

Reply:
Well, I don't know about K8s, but it's what my local Fedora system uses ^^